OrbitGl/PickingManager.cpp

```diff
@@ -8,52 +8,77 @@
 
 //-----------------------------------------------------------------------------
 PickingID PickingManager::CreatePickableId(Pickable* a_Pickable) {
+  absl::MutexLock lock(&mutex_);
   ++m_IdCounter;
-  PickingID id = PickingID::Get(PickingID::PICKABLE, ++m_IdCounter);
+  PickingID id = PickingID::Get(PickingID::PICKABLE, m_IdCounter);
   m_PickableIdMap[a_Pickable] = m_IdCounter;
   m_IdPickableMap[m_IdCounter] = a_Pickable;
   return id;
 }
 
 //-----------------------------------------------------------------------------
-void PickingManager::ClearIds() {
+void PickingManager::Reset() {
+  absl::MutexLock lock(&mutex_);
   m_IdPickableMap.clear();
   m_PickableIdMap.clear();
+  m_IdCounter = 0;
+}
+
+//-----------------------------------------------------------------------------
+Pickable* PickingManager::GetPickableFromId(uint32_t id) {
+  absl::MutexLock lock(&mutex_);
+  return m_IdPickableMap[id];
```

Comment:
Is it ok if "id" is not yet an existing key here? This adds a new element to the map, is this intended? (If it can never happen with normal usage, I would add a check / assert here that tests if the id exists in the map.)

Reply:
It is actually OK.  In the case where we inadvertently draw primitives using a color that is wasn't generated as a Picking Color, when we ask for the corresponding Pickable, it will return a nullptr.  We should probably warn in that case.  I'll add that.

Reply:
In the end I chose not to warn as this happens any time we click at a location where there is no "pickable".  I did however made it so that we don't add the nullptr to the map but rather return nullptr explicitly.